File: 프로그래머스/2/49993. 스킬트리/스킬트리.js
function solution(skill, skill_trees) {
    skill = [...skill]
    let cnt = skill_trees.length
    
    for(let i = 0; i < skill_trees.length; i++) {
        const stack = []
        for(let j = 0; j < skill_trees[i].length; j++) {
            if(skill.includes(skill_trees[i][j])) {
                stack.push(skill_trees[i][j])
            }
        }
        
        for(let j = 0; j < stack.length; j++) {
            if(stack[j] !== skill[j]) {
                cnt--
                break
            }
        }
    }
    
    return cnt
    
}

File: KimChaeJung/프로그래머스/lv2/49993. 스킬트리/스킬트리.py
def solution(skill, skill_trees):
    answer = 0
    skillArr = list(skill)
    for ex in skill_trees:
        check = skillArr.copy()
        invalid = False
        for single in ex:
            if single in check:
                if single == check[0]:
                    check.pop(0)
                else:
                    invalid = True
                    break
        if len(check) == 0:
            answer += 1
        elif not invalid:
            answer += 1
                
    return answer